Installer fails on amd strix halo

### Description

1. cloned fresh repo
2. executed ./installer.sh
3. Received error:

[ERROR] Generated .env failed schema validation. See /tmp/dream-server-install.log for details.  

4. Log file: 

[ERROR] Unknown keys not defined in schema:                                                                                  
  - LLAMA_CPP_REF (line 42)                                                                                                  
  - HSA_XNACK (line 39)                                                                                                      
  - AMDGPU_TARGET (line 41)   

5. uname -a

Linux #### 6.17.0-1017-oem #17-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 27 13:48:03 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux

6. cat /etc/os-release 
PRETTY_NAME="Ubuntu 24.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.4 LTS (Noble Numbat)"

7. amd-smi 
+------------------------------------------------------------------------------+
| AMD-SMI 26.2.1+fc0010cf6a    amdgpu version: Linuxver ROCm version: 7.2.0    |
| VBIOS version: 023.011.000.039.000001                                        |
| Platform: Linux Baremetal            

8. Hardware: Bosgame M5 strix halo 128gb gfx1151
9. 



### Steps to Reproduce

1. Run ./install.sh
2. Select option 1

### Expected Behavior

Would expect installation to succeed

### Actual Behavior

Install failed

### Operating System

Ubuntu 24.04

### GPU

AMD strix halo gfx1151

### Docker Version

none

### VRAM

110 Gb allocated

### Logs

```shell
[0;32m[INFO][0m Detected distro: ubuntu (like: debian, pkg: apt)            
    [0;32m▸[0m I'm scanning your system for required components...            
  [0;32m[INFO][0m Detected OS: Ubuntu 24.04.4 LTS                             
  [0;32m[INFO][0m curl: curl 8.5.0 (x86_64-pc-linux-gnu) libcurl/8.5.0        
  OpenSSL/3.0.13 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.7 libpsl/0.21.2 
  (+libidn2/2.3.7) libssh/0.10.6/openssl/zlib nghttp2/1.59.0 librtmp/2.3      
  OpenLDAP/2.6.10                                                             
  [0;32m[INFO][0m jq: jq-1.7                                                  
    [1;32m✓[0m Pre-flight checks passed.                                      
    [0;32m░▒▓█▓▒░[0m No cloud dependencies required for core operation.       
    [0;32m▸[0m Reading hardware telemetry...                                  
  [0;32m[INFO][0m Capability profile loaded: /tmp/dream-server-capabilities.  
  json                                                                        
  [0;32m[INFO][0m Capability profile: platform=linux, gpu=amd, tier=SH_LARGE  
  [0;32m[INFO][0m Hardware class: strix_halo_395 (AMD Ryzen AI MAX+ 395 (Strix
 [0;32m[INFO][0m Capability profile: platform=linux, gpu=amd, tier=SH_LARGE                              10:20:38 [52/41144]
  [0;32m[INFO][0m Hardware class: strix_halo_395 (AMD Ryzen AI MAX+ 395 (Strix
  Halo))                                                                      
  [0;32m[INFO][0m RAM: 124GB                                                  
  [0;32m[INFO][0m Available disk: 818GB                                       
    [0;32m▸[0m Detecting GPU...                                               
  [0;32m[INFO][0m GPU: AMD APU (0x1586) (unified memory, AMD APU,             
  device_id=0x1586)                                                           
  [0;32m[INFO][0m Capabilities override detection: backend=amd, memory=unified,
  tier=SH_LARGE                                                               
  [0;32m[INFO][0m Backend contract loaded: /home/ncuka/git/DreamServer/dream- 
  server/config/backends/amd.json                                             
  [0;32m[INFO][0m Backend runtime: amd (lemonade)                             
  [0;32m[INFO][0m Auto-detected tier: SH_LARGE                                
  [0;32m[INFO][0m Compose selection: -f docker-compose.base.yml -f docker-    
  compose.amd.yml -f extensions/services/ape/compose.yaml -f                  
  extensions/services/comfyui/compose.yaml -f                                 
  extensions/services/comfyui/compose.amd.yaml -f                             
  extensions/services/dashboard/compose.local.yaml -f                         
  extensions/services/dreamforge/compose.yaml -f                              
  extensions/services/embeddings/compose.yaml -f                              
  extensions/services/litellm/compose.yaml -f                                 
  extensions/services/litellm/compose.amd.yaml -f      
 extensions/services/litellm/compose.yaml -f                                                             10:20:38 [31/41144]
  extensions/services/litellm/compose.amd.yaml -f                             
  extensions/services/litellm/compose.local.yaml -f                           
  extensions/services/n8n/compose.yaml -f extensions/services/open-           
  webui/compose.local.yaml -f extensions/services/openclaw/compose.yaml -f    
  extensions/services/openclaw/compose.amd.yaml -f                            
  extensions/services/openclaw/compose.local.yaml -f                          
  extensions/services/perplexica/compose.yaml -f                              
  extensions/services/perplexica/compose.local.yaml -f                         
  extensions/services/privacy-shield/compose.yaml -f                          
  extensions/services/privacy-shield/compose.local.yaml -f                    
  extensions/services/qdrant/compose.yaml -f                                  
  extensions/services/searxng/compose.yaml -f extensions/services/token-      
  spy/compose.yaml -f extensions/services/tts/compose.yaml -f                 
  extensions/services/whisper/compose.yaml                                    
    [0;32m▸[0m Choose how deep you want to go. I can install everything, or   
  keep it minimal.                                                            
    [0;32m░▒▓█▓▒░[0m Acknowledged.                                            
  [0;32m[INFO][0m Selected: Full Stack                                        
  [0;32m[INFO][0m Compose flags refreshed after feature selection             
  [0;32m[INFO][0m OpenClaw config: openclaw-strix-halo.json (matched to Tier  
  SH_LARGE)                                                                   
 [0;32m[INFO][0m All services enabled (core install)                         
  [0;32m[INFO][0m Single GPU detected — non-NVIDIA backend, skipping GPU      
  assignment.                                                                 
  [0;32m[INFO][0m Preflight report: /tmp/dream-server-preflight-report.json   
    [1;32m✓[0m Preflight passed with 0 warning(s).                            
    [0;32m▸[0m Preparing container runtime...                                 
    [1;32m✓[0m Docker already installed: Docker version 29.4.0, build 9d7ad9f 
    [1;32m✓[0m Docker Compose v2 available                                     
  [0;32m[INFO][0m Using docker command: docker                                
    [1;32m✓[0m Compose detected: docker compose                               
    [1;32m✓[0m Docker engine responding                                       
    [0;32m▸[0m Copying source files to /home/ncuka/dream-server...            
    [1;32m✓[0m Source files installed                                         
    [1;32m✓[0m Extensions library copied to data/extensions-library/          
  [0;32m[INFO][0m Installed OpenClaw config: openclaw-strix-halo.json ->      
  openclaw.json (model: qwen3-coder-next)                                     
  [0;32m[INFO][0m Installed OpenClaw workspace files (agent personality)      
    [1;32m✓[0m Created /home/ncuka/dream-server                               
    [1;32m✓[0m Generated secure secrets in .env (permissions: 600)            
    [1;32m✓[0m Generated LiteLLM config for Lemonade (model: extra.Qwen3.5-2B-
  Q4_K_M.gguf)    
[0;31m[ERROR][0m Unknown keys not defined in schema:                        
    - LLAMA_CPP_REF (line 42)                                                 
    - HSA_XNACK (line 39)                                                     
    - AMDGPU_TARGET (line 41)                                                 
                                                                              
  [0;34m[INFO][0m Fix .env using .env.example as reference, then re-run:      
    ./scripts/validate-env.sh                                                 
  [0;31m[ERROR][0m Generated .env failed schema validation. See /tmp/dream-   
  server-install.log for details.
```

### Screenshots

_No response_

--- a/dream-server/.env.schema.json
+++ b/dream-server/.env.schema.json
@@ -333,6 +333,18 @@
       "type": "integer",
       "description": "AMD ROCm BLAS setting"
     },
+    "HSA_XNACK": {
+      "type": "integer",
+      "description": "AMD ROCm extended memory access for large KV caches"
+    },
+    "AMDGPU_TARGET": {
+      "type": "string",
+      "description": "GPU arch for llama.cpp build (gfx1151, gfx1100, etc.)"
+    },
+    "LLAMA_CPP_REF": {
+      "type": "string",
+      "description": "llama.cpp release tag to build (pin to avoid breakage)"
+    },
     "UID": {
       "type": "integer",
       "description": "Container user ID",

--- a/dream-server/tests/contracts/test-installer-contracts.sh
+++ b/dream-server/tests/contracts/test-installer-contracts.sh
@@ -30,6 +30,12 @@ echo "[contract] capability profile schema has hardware_class"
 jq -e '.properties.hardware_class and (.required | index("hardware_class"))' config/capability-profile.schema.json >/dev/null \
   || { echo "[FAIL] capability profile schema missing hardware_class"; exit 1; }
 
+echo "[contract] AMD phase-06 env keys exist in schema"
+for key in HSA_XNACK AMDGPU_TARGET LLAMA_CPP_REF; do
+  jq -e --arg key "$key" '.properties[$key]' .env.schema.json >/dev/null \
+    || { echo "[FAIL] .env.schema.json missing AMD installer key: $key"; exit 1; }
+done
+
 echo "[contract] canonical port contract parity"
 test -x tests/contracts/test-port-contracts.sh || { echo "[FAIL] script not executable: tests/contracts/test-port-contracts.sh"; exit 1; }
 bash tests/contracts/test-port-contracts.sh

